AstSemantics.md

```diff
@@ -142,11 +143,12 @@ which leads to the following advantages:
 
 ## Accessing the heap
 
-Each heap access is annotated with a *Memory type* and
-the presumed alignment of the incoming pointer. As discussed previously, loads may
-include explicit zero- or sign-extension and stores may include implicit wrapping.
-
-Indexes into the heap are always byte indexes.
+Programs address the heap by using integers that are interpreted as unsigned byte indexes
+starting at `0`.
+Accesses to the heap at indices larger than the size of the heap are considered out-of-bounds,
+and a module may optionally define that out-of-bounds includes small indices close to `0`.
```

Comment:
I think it's worth discussing how this works. I like the idea, it would be good to define it further. Could you add a link to #204 so it's clear that this is still TBD?